File: src/main/java/io/github/tanguygab/conditionalactions/DataManager.java
package io.github.tanguygab.conditionalactions;

import org.bukkit.OfflinePlayer;

import java.util.HashMap;
import java.util.Map;

public class DataManager {

    private final Map<OfflinePlayer,Map<String,String>> data = new HashMap<>();
    private final Map<String,String> globalData = new HashMap<>();

    public DataManager() {
        // load data.yml
    }

    public boolean isGlobal(String name) {
        return name.equalsIgnoreCase("--global");
    }

    public void setData(OfflinePlayer player, String data, String value) {
        this.data.computeIfAbsent(player,k->new HashMap<>()).put(data,value);
    }
    public void removeData(OfflinePlayer player, String data) {
        this.data.computeIfAbsent(player,k->new HashMap<>()).remove(data);
    }

    public void setGlobalData(String data, String value) {
        globalData.put(data,value);
    }
    public void removeGlobalData(String data) {
        globalData.remove(data);
    }

    public String getData(OfflinePlayer player, String data) {
        return this.data.getOrDefault(player,Map.of()).getOrDefault(data,"");
    }
    public String getGlobalData(String data) {
        return globalData.getOrDefault(data,"");
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/DataManager.kt
package io.github.tanguygab.conditionalactions

import org.bukkit.OfflinePlayer

class DataManager {
    private val data = mutableMapOf<OfflinePlayer, MutableMap<String, String>>()
    private val globalData = mutableMapOf<String, String>()

    fun isGlobal(name: String) = name.equals("--global", ignoreCase = true)

    fun setData(player: OfflinePlayer, data: String, value: String) {
        this.data.computeIfAbsent(player) { mutableMapOf() }[data] = value
    }
    fun removeData(player: OfflinePlayer, data: String) {
        this.data.computeIfAbsent(player) { mutableMapOf() }.remove(data)
    }

    fun setGlobalData(data: String, value: String) = globalData.put(data, value)
    fun removeGlobalData(data: String) = globalData.remove(data)

    fun getData(player: OfflinePlayer, data: String) = (this.data[player] ?: mutableMapOf())[data] ?: ""
    fun getGlobalData(data: String?) = globalData[data] ?: ""
}